crates/e2e/tests/e2e/solver_competition.rs

```diff
@@ -189,8 +195,143 @@ async fn fairness_check(web3: Web3) {
+
+    // Verify that test_solver was excluded due to fairness check
+    let trades = services.get_trades(&uid_a).await.unwrap();
+    let competition = services
+        .get_solver_competition(trades[0].tx_hash.unwrap())
+        .await
+        .unwrap();
+    tracing::info!(?competition, "competition");
+    assert_eq!(
+        competition.common.solutions.last().unwrap().solver,
+        "solver2"
+    );
+    assert_eq!(competition.common.solutions.len(), 1);
+}
+
+async fn wrong_solution_submission_address(web3: Web3) {
+    let mut onchain = OnchainComponents::deploy(web3).await;
+
+    let [solver] = onchain.make_solvers(to_wei(1)).await;
+    let [trader_a, trader_b] = onchain.make_accounts(to_wei(1)).await;
+    let [token_a, token_b] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(1_000), to_wei(1_000))
+        .await;
+
+    // Fund traders
+    token_a.mint(trader_a.address(), to_wei(10)).await;
+    token_b.mint(trader_b.address(), to_wei(10)).await;
+
+    // Create more liquid routes between token_a (token_b) and weth via base_a
+    // (base_b). base_a has more liquidity then base_b, leading to the solver that
+    // knows about base_a to win
+    let [base_a, base_b] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(10_000), to_wei(10_000))
+        .await;
+    onchain
+        .seed_uni_v2_pool((&token_a, to_wei(100_000)), (&base_a, to_wei(100_000)))
+        .await;
+    onchain
+        .seed_uni_v2_pool((&token_b, to_wei(10_000)), (&base_b, to_wei(10_000)))
+        .await;
+
+    // Approve GPv2 for trading
+    tx!(
+        trader_a.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(100))
+    );
+    tx!(
+        trader_b.account(),
+        token_b.approve(onchain.contracts().allowance, to_wei(100))
+    );
+
+    // Start system, with two solvers, one that knows about base_a and one that
+    // knows about base_b
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![
+            colocation::start_baseline_solver(
+                "test_solver".into(),
+                solver.clone(),
+                onchain.contracts().weth.address(),
+                vec![base_a.address()],
+            )
+            .await,
+            colocation::start_baseline_solver(
+                "solver2".into(),
+                solver.clone(),
+                onchain.contracts().weth.address(),
+                vec![base_b.address()],
+            )
+            .await,
+        ],
+        colocation::LiquidityProvider::UniswapV2,
+    );
+
+    let services = Services::new(&onchain).await;
+    services.start_autopilot(
+        None,
+        // Solver 1 has a wrong submission address, meaning that the solutions should be discarded from solver1
+        vec![
+            format!("--drivers=solver1|http://localhost:11088/test_solver|0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2,solver2|http://localhost:11088/solver2|{}", hex::encode(solver.address())),
```

Comment:
I think there is a wrong comment on line 396 (not visible here). I assume the point is that solution was excluded due to solution being filtered because of wrong submission address and not fairness